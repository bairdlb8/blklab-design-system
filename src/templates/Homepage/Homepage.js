import React from 'react'
import PropTypes from 'prop-types'

import './Homepage.scss'
import GridContainer from '../../containers/GridContainer/GridContainer'
import FlexContainer from '../../containers/FlexContainer/FlexContainer'
import Container from '../../containers/Container/Container'
import Layout from '../../layouts/Layout/Layout'
import Fpo from '../../basics/Fpo/Fpo'
import Tag from '../../basics/Tag/Tag'

const Homepage = ({ className = '' }) => {
  return (
    <Layout>
      <Container>
        <Fpo>Hero</Fpo>
        <GridContainer gridSize={2} gridGap={20}>
          <Fpo>About Block</Fpo>
          <Fpo>Another Block</Fpo>
        </GridContainer>
        <FlexContainer>
          <Tag>Test</Tag>
          <Tag>Other</Tag>
        </FlexContainer>
      </Container>
    </Layout>
  )
}

Homepage.propTypes = {}

export default Homepage
