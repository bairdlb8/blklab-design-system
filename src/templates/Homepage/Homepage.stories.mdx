import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import Homepage from './Homepage'

<Meta title="Templates / Homepage" component={Homepage} />

# Homepage

## Add some text here

export const Template = (args) => {
  return <Homepage {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={{}}>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={Homepage} />
