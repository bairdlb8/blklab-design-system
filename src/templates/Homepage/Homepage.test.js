import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { toHaveNoViolations } from 'jest-axe'
import Homepage from './Homepage'

expect.extend(toHaveNoViolations)

describe('<Homepage />', () => {

  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<Homepage />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<Homepage />)

    expect(asFragment(<Homepage />)).toMatchSnapshot()
  })

})
