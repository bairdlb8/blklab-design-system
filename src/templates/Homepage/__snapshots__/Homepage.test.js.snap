// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> should take a snapshot 1`] = `
<DocumentFragment>
  <main>
    Hi
  </main>
</DocumentFragment>
`;
